scripts/hotspot.py

```diff
@@ -0,0 +1,148 @@
+    else:
+        return args
+
+"""Small sanity check for two different feature fields
+
+Args:
+    csq (str): string denoting the CSQ field
+    hotspot (str): a hotspot
+
+Return:
+    str: Info about the CSQ field and hotspot
+"""
+def csq_sanity(field,value):
+    if (field == "Gene" and not value.startswith("ENSG")) or (field == "Feature" and not value.startswith("ENST")):
+        return "CSQ field: {} potentially mismatches hotspot format: {}".format(field,value)
+    else:
+        return "CSQ field: {}; Example hotspot: {}".format(field,value)
+
+"""Function to read the hotspots file and return a list of unique hotspots
+
+Args:
+    infile (file): file containing hotspot information
+    csq (str): csq
+
+Return:
+    list: returns a list of unique hotspot values from the file
+"""
+def process_hotspots(infile,field):
+    hs_list = []
+    with open(infile) as fh:
+        rd = csv.reader(fh, delimiter='\t')
+        for row in rd:
+            if row[0].startswith('#'):
+                continue
+            hs_list.append(row[3].split(','))
+    unique_hs = list(set([items for sublist in hs_list for items in sublist]))
+    print(csq_sanity(field,unique_hs[0]))
+    return unique_hs
+
+"""Function to take the input VCF and add INFO annotations for the ALT alleles that cause protein changes at hotspots
+
+Args:
+    infile (file): Path to the VCF(.GZ) file
+    outfile (str): String name for the output file that will be made in the fuction
+    hslist (list of strings): Unique list of hotspots derived from the provided input in the format <CSQ_FIELD>:<Amino_acids[ref]><Protein_position[start]>
+    fid (str): the name of the CSQ_FIELD that is being used as the ID in the hslist; commonly Hugo Symbols: SYMBOL, Ensembl Gene IDs: Gene, or Ensembl Transcrip IDs: Feature
+    aa (str): the name of the CSQ_FIELD that contains the amino acid change annotation
+    pos (str): the name of the CSQ_FIELD that contains the protein position annotation
+    allele (str): the name of the CSQ_FIELD that contains the allele number annotation
+
+Return:
+    file: A VCF.GZ file is written to Args.outfile
+    int: total_hotspots_found The total number of hotspot annotations in the file
+"""
+def process_vcf(infile,outfile,hslist,fid,aa='Amino_acids',pos='Protein_position',allele='ALLELE_NUM'):
+    vcfin = pysam.VariantFile(infile)
+    vcfin.header.info.add("HotSpotAllele",'A','Integer',"Is the corresponding ALT allele on a hotspot (0 for false; 1 for true).")
+    vcfout = pysam.VariantFile(outfile,'w',header=vcfin.header)
+    csq_items = vcfin.header.info["CSQ"].description.split(' Format: ')[1].split('|')
+    total_hotspots_found = 0
+    for rec in vcfin.fetch():
+        hotspot_allele_nums = set([i.split('|')[csq_items.index(allele)] for i in rec.info['CSQ'] if '{}:{}{}'.format(i.split('|')[csq_items.index(fid)],i.split('|')[csq_items.index(aa)].split('/')[0],i.split('|')[csq_items.index(pos)].split('/')[0]) in hslist])
```

Comment:
While I have a grudging respect for this massive one-liner, it is very hard to tell what it does.

Reply:
I'm keeping it, but I did put a more traditional, commented-out for loop beneath it so people can look at that and understand what's going on here. Once you understand, it's really not complicated at all. It's mostly just intimidating because of all the splitting.